### ISSUE TITLE
How to get image that was uploaded ?

### ISSUE THREAD

**Issue by netrij** [kolkov/angular-editor#39]
Uploading is working but how to get that image back in editor. 
In **uploadUrl: 'api',** , what to return in that api, url of image or what ? I tried to return that url of image but image is not showing !

**Comment by kolkov** [comment]
Hi! Try to read this issue.
https://github.com/kolkov/angular-editor/issues/5

**Comment by dchirutac** [comment]
Hi @kolkov, as the @netrij said, there is no solution to list the uploaded images from the server. It will be great to have an **AngularEditorConfig** property where we can enter an endpoint or a list of urls listed from the server.

**Comment by kolkov** [comment]
Hi! If I understand correctly, you want to receive from the server a list of downloaded pictures with urls?

**Comment by dchirutac** [comment]
Yes, in my case I have on the server in a specific path directory all the images ready to use.

**Comment by kolkov** [comment]
and in the editor you want a pop-up window, where there will be a preview of the uploaded pictures, for insertion?

**Comment by dchirutac** [comment]
Hi @kolkov, any updates ?

**Comment by nunos7** [comment]
Hi, i upload a image to the backend, and return a json object like this:
`result.msg = "upload OK"
result.imageUrl = "../images/myimage.jpg"
result.status = true
`
C# UploadController
`
[HttpPost("UploadBase64")]
        public IActionResult UploadBase64(IFormFile file)
        {
            try
            {              
                // save o disk
               // code to save here!!!!
               // ************************

                OperationResult result = new OperationResult();                               
                result.status = true;
                result.msg = "upload OK!";
                result.imageUrl = imgPath;

                return new JsonResult(result);
            }
            catch (Exception ex)
            {
                OperationResult result = new OperationResult();
                result.status = false;
                result.msg = ex.ToString();

                return new JsonResult(result);
            }
        }
`

But the image is not load on the editor! 
Some help, please?

**Comment by nunos7** [comment]
If we click on the upload image, before giving the focus to the editor, the image is uploaded but the image is not loaded in the editor. We have to first focus on the editor and then click on the button to insert the image.

**Comment by MPOscar** [comment]
![comment](https://user-images.githubusercontent.com/35536348/56371285-da950c80-61ca-11e9-8ca2-33f9b27a21d4.png)
for node api i have to add this and recompile the library

**Comment by prakashksp20** [comment]
> Uploading is working but how to get that image back in editor.
> In **uploadUrl: 'api',** , what to return in that api, url of image or what ? I tried to return that url of image but image is not showing !

Yes, I'm also face same issue. Anyone have a solution, kindly update it

**Comment by Admiraal** [comment]
In my opinion it's much more useful to be able to respond to an upload event. So one can simply handle the upload in the clientcomponent and decise which service to use from there. 
As I would like to store the file on firestore, an api address is not applicable.

**Comment by sfaxianovic** [comment]
Hi all,
Thanks for this component.
I have a question why reading the image using document.executeCommand and not using angular http client.

Using http client to get the image allow us to use interceptor to add header ?
Can we have this enhancement soon, if we submit a pull request ?

**Comment by tuku-dev** [comment]
> If we click on the upload image, before giving the focus to the editor, the image is uploaded but the image is not loaded in the editor. We have to first focus on the editor and then click on the button to insert the image.

I have tried this but it ended up with a blank <img> tag.

**Comment by kolkov** [comment]
This question has been extensively answered in the 14 comments!

**The answer:** Your upload API must return JSON with `imageUrl` property:

```json
{
  "imageUrl": "https://example.com/uploads/image123.jpg"
}
```

**Backend example:**
Your server endpoint should return the uploaded image URL in the response body with an `imageUrl` property.

**Or use custom upload function:**

```typescript
editorConfig = {
  upload: (file: File) => {
    const formData = new FormData();
    formData.append('image', file);
    
    return this.http.post('/api/upload', formData).pipe(
      map((response: any) => new HttpResponse({
        body: { imageUrl: response.url }
      }))
    );
  }
};
```

The editor automatically inserts the image using the `imageUrl` from your API response.

Closing as this has been thoroughly answered in comments. For further upload questions, please use [GitHub Discussions](https://github.com/kolkov/angular-editor/discussions)!
